pymeasure/adapters/adapter.py

```diff
@@ -114,6 +119,29 @@ def binary_values(self, command, header_bytes=0, dtype=np.float32):
         raise NameError("Adapter (sub)class has not implemented the "
                         "binary_values method")
 
+    def __getattribute__(self, name):
+        """ Override to acquire the 'connection_lock' lock through the duration of attribute access.
+        """
+        if name not in ("connection_lock", "connection_lock_initialized") \
```

Comment:
I'm curious about the choice to intercept _all attributes_, when the stated intention (and lock name) indicate that you only want to lock on _connection_ access. I'd be more comfortable with a tighter-scoped version where you only put a lock around `self.connection`, which is what we care about. 
Otherwise, you are also locking e.g. `PrologixAdapter.address`, which does not make sense.